Main/ClientApp/src/Authorization/Logout.js

```diff
@@ -1,128 +1,152 @@
-                case LogoutActions.LoggedOut:
-                    return (<div>{message}</div>);
-                default:
-                    throw new Error(`Invalid action '${action}'`);
-            }
+          // This prevents regular links to <app>/authentication/logout from triggering a logout
+          this.setState({
+            isReady: true,
+            message: "The logout was not initiated from within the page."
+          });
         }
+        break;
+      case LogoutActions.LogoutCallback:
+        this.processLogoutCallback();
+        break;
+      case LogoutActions.LoggedOut:
+        this.setState({
+          isReady: true,
+          message: "You successfully logged out!"
+        });
+        break;
+      default:
+        throw new Error(`Invalid action '${action}'`);
     }
 
-    async logout(returnUrl) {
-        const state = { returnUrl };
-        const isauthenticated = await authService.isAuthenticated();
-        if (isauthenticated) {
-            const result = await authService.signOut(state);
-            switch (result.status) {
-                case AuthenticationResultStatus.Redirect:
-                    break;
-                case AuthenticationResultStatus.Success:
-                    await this.navigateToReturnUrl(returnUrl);
-                    break;
-                case AuthenticationResultStatus.Fail:
-                    this.setState({ message: result.message });
-                    break;
-                default:
-                    throw new Error("Invalid authentication result status.");
-            }
-        } else {
-            this.setState({ message: "You successfully logged out!" });
-        }
-    }
+    this.populateAuthenticationState();
+  }
 
-    async processLogoutCallback() {
-        const url = window.location.href;
-        const result = await authService.completeSignOut(url);
-        switch (result.status) {
-            case AuthenticationResultStatus.Redirect:
-                // There should not be any redirects as the only time completeAuthentication finishes
-                // is when we are doing a redirect sign in flow.
-                throw new Error('Should not redirect.');
-            case AuthenticationResultStatus.Success:
-                await this.navigateToReturnUrl(this.getReturnUrl(result.state));
-                break;
-            case AuthenticationResultStatus.Fail:
-                this.setState({ message: result.message });
-                break;
-            default:
-                throw new Error("Invalid authentication result status.");
-        }
+  render() {
+    const { isReady, message } = this.state;
+    if (!isReady) {
+      return <div></div>;
+    }
+    if (!!message) {
+      return <Center>{message}</Center>;
+    } else {
+      const action = this.props.action;
+      switch (action) {
+        case LogoutActions.Logout:
+          return (
+            <Center>
```

Comment:
Looks like there is a lot of redundant <Center> components here. Maybe the whole Logout should be wrapped in the <Center> instead